packages/react/src/PricingOptions/PricingOptions.tsx

```diff
@@ -0,0 +1,479 @@
+        if (React.isValidElement(child) && typeof child.type !== 'string') {
+          if (child.type === PricingOptionsFeatureList) {
+            acc.FeatureList = child as React.ReactElement<PricingOptionsFeatureListProps> | null
+          }
+
+          if (child.type === PricingOptionsPrimaryAction || child.type === PricingOptionsSecondaryAction) {
+            acc.Actions.push(child as React.ReactElement<PricingOptionsActionsProps>)
+          }
+
+          if (child.type === PricingOptionsFootnote) {
+            acc.Footnote = child as React.ReactElement<PricingOptionsFootnoteProps>
+          }
+
+          if (
+            child.type === PricingOptionsLabel ||
+            child.type === PricingOptionsHeading ||
+            child.type === PricingOptionsDescription ||
+            child.type === PricingOptionsPrice
+          ) {
+            acc.Content.push(
+              child as React.ReactElement<
+                | PricingOptionsLabelProps
+                | PricingOptionsDescriptionProps
+                | PricingOptionsHeadingProps
+                | PricingOptionsPriceProps
+              >,
+            )
+          }
+        }
+        return acc
+      },
+      {Content: [], FeatureList: null, Actions: [], Footnote: null},
+    )
+
+    return (
+      <div
+        className={clsx(styles.PricingOptions__item, className)}
+        data-testid={testId || testIds.item}
+        ref={ref}
+        {...(rest as HTMLAttributes<HTMLElement>)}
+      >
+        {Content}
+        {Actions.length > 0 && <div className={styles.PricingOptions__actions}>{Actions}</div>}
+        {FeatureList}
+        {Footnote}
+      </div>
+    )
+  },
+)
+
+type PricingOptionsLabelProps = PropsWithChildren<BaseProps<HTMLSpanElement>> & {
+  'data-testid'?: string
+}
+
+const PricingOptionsLabel = forwardRef<HTMLSpanElement, PricingOptionsLabelProps>(
+  ({children, className, 'data-testid': testId, ...rest}, ref) => {
+    return (
+      <Label
+        className={clsx(styles.PricingOptions__label, className)}
+        data-testid={testId || testIds.label}
+        ref={ref}
+        size="medium"
+        {...rest}
+      >
+        {children}
+      </Label>
+    )
+  },
+)
+
+type PricingOptionsDescriptionProps = PropsWithChildren<BaseProps<HTMLParagraphElement>> & {
+  'data-testid'?: string
+}
+
+const PricingOptionsDescription = forwardRef<HTMLParagraphElement, PricingOptionsDescriptionProps>(
+  ({children, className, 'data-testid': testId, ...rest}, ref) => {
+    return (
+      <Text
+        as="p"
+        className={clsx(styles.PricingOptions__description, className)}
+        data-testid={testId || testIds.description}
+        ref={ref}
+        size="200"
+        variant="muted"
+        {...rest}
+      >
+        {children}
+      </Text>
+    )
+  },
+)
+
+type PricingOptionsHeadingProps = BaseProps<HTMLHeadingElement> & {
+  as?: Exclude<HeadingProps['as'], 'h1' | 'h2'>
+  children: React.ReactNode | React.ReactNode[]
```

Comment:
Could we use PropsWithChildren for this like you did in the other sub components?